### DISCUSSION TITLE
Fog effect

### DISCUSSION THREAD

**Discussion by mogmog** [visgl/deck.gl#4458] [Q&A]
I am working with someone who is experienced in WebGL but not Deck specifically.

We are trying to achieve a fog effect, in the same way that Caesium achieves. Maybe even with the ability to control properties of the fog.

We have had some preliminary success, but has involved a lot of additional GL code, specifically in onWebGLInitialized to access the depth buffer after draw.

Is this additional onWebGLInitialized code required or does DeckGL expose an api for such things? 

If any one can suggest a way forward I am happy to share the code, with the intention of creating an fog effect in addition to the lighting effects.

**Comment by xintongxia** [comment]
Deck.gl support customized effects, here is the [doc](https://deck.gl/#/documentation/deckgl-api-reference/effects/post-process-effect?section=constructor)

**Comment by mogmog** [comment]
I might be wrong as I am translating from someone else, but I have been advised that "Using postprocess you can't get depth frame buffer, because it does not exist, so that's why i'm creating it onWebGLInitialized"
